scripts/hotfixes/kdump/create-kdump-artifacts.sh

```diff
@@ -0,0 +1,258 @@
+#!/bin/bash
+#
+# MIT License
+#
+# (C) Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+set -euo pipefail
+
+# Source common dracut parameters.
+. "$(dirname $0)/dracut-lib.sh"
+
+# show the line that we failed and on and exit non-zero
+
+trap 'catch $? $LINENO; cleanup; exit 1' ERR
+
+# if the script is interrupted, run the cleanup function
+trap 'cleanup' INT EXIT
+
+# catch() prints what line the script failed on, runs the cleanup function, and then exits non-zero
+catch() {
+    # Show what line the error occurred on because it can be difficult to detect in a dracut environment
+    echo "CATCH: exit code $1 occurred on line $2 in $(basename "${0}")"
+    cleanup
+    exit 1
+}
+
+
+# cleanup() removes temporary files, puts things back where they belong, etc.
+cleanup() {
+    echo "CLEANUP: cleanup function running ..."
+
+    # Restore the dracut config if it was removed.
+    [ ! -f /etc/dracut.conf.d/05-metal.conf ] && cp -v /run/rootfsbase/etc/dracut.conf.d/05-metal.conf /etc/dracut.conf.d/05-metal.conf
+
+    # Stops re-running this script on reboot
+    # TODO: This script should move to the pipeline, and the service can then be removed (MTL-1830).
+    systemctl disable kdump-cray
+}
+
+
+# check_size() offers a CAUTION message if the initrd is larger then 20MB
+# this is just a soft warning since several factors can influence running out of memory including:
+# crashkernel= parameters, drivers that are loaded, modules that are loaded, etc.
+# so it's more of a your mileage may vary message
+check_size() {
+
+    local initrd="$1"
+    local max=20000000 # kdump initrds larger than 20M may run into issues with memory
+    
+    if [[ "$(stat --format=%s $initrd)" -ge "$max" ]]; then
+        echo >&2 "CAUTION: initrd might be too large ($(stat --format=%s $initrd)) and may OOM if used"
```

Comment:
I think it's safer not to assume the person running the script will realize what is meant by OOM. I'd change the wording slightly.

```suggestion
        echo >&2 "CAUTION: initrd might be too large ($(stat --format=%s $initrd)) and may exceed available memory (OOM) if used"
```

Reply:
I just merged the original PR, so I am going to leave this.

Reply:
If you just merged it, it seems like it wouldn't be tough to make a follow up PR to make edits, unless we're cranking out a new build right this moment or something...

Reply:
Or put another way, if it was important for the two to be common, probably neither PR should merge until both had the approvals. It feels like a way to short-circuit the review process otherwise. Not that these issues are life or death, but it feels odd to deliberately not fix minor things.

Reply:
No it's just Friday and it was sitting for a long time. I had incorporated other comments and didn't see your new comments.

Reply:
I figured you wouldn't have new comments about that script being you had reviewed this already, and then I had merged it before seeing your new new comments.

Reply:
I'll make a PR to include this one.

Reply:
https://github.com/Cray-HPE/metal-provision/pull/16